python/lsst/daf/butler/_butlerRepoIndex.py

```diff
@@ -0,0 +1,173 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("ButlerRepoIndex",)
+
+import os
+
+from typing import (
+    ClassVar,
+    Dict,
+    Set,
+)
+
+from .core import (
+    ButlerURI,
+    Config,
+)
+
+
+class ButlerRepoIndex:
+    """Index of all known butler repositories.
+
+    The index of butler repositories is found by looking for a
+    configuration file at the URI pointed at by the environment
+    variable ``$BUTLER_REPOSITORY_INDEX``. The configuration file
```

Comment:
Should this have an `LSST_` prefix to help avoid collisions, even if the Butler is being reused outside the project?

Reply:
Okay. So it looks like butler currently recognizes the following environment variables:

* DAF_BUTLER_PLUGINS
* DAF_BUTLER_CONFIG_PATH
* DAF_BUTLER_CACHE_DIRECTORY
* DAF_BUTLER_CACHE_EXPIRATION_MODE
* DAF_BUTLER_TEST_TMP (for test code)
* LSST_WEBDAV_*. (for ButlerURI webdav implementation)

So you make a strong case for DAF_BUTLER_REPOSITORY_INDEX...